src/service/data_products/heatmap_common_models.py

```diff
@@ -72,10 +90,12 @@ class Cell(BaseModel):
         example="8",
         description="The ID of the column in which this cell is located."
     )
-    val: float = Field(
+    val: float | bool = Field(
         example=4.2,
         description="The value of the heatmap at this cell."
     )
+    class Config:
+        smart_union=True
```

Comment:
I am not exactly sure what does that do.

Reply:
It makes the union types (like `float | bool`) work as you expect. Otherwise they always get coerced to the first type in the union. I'm guessing it's a backwards compatibility thing

Reply:
👍